File: code/backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/commun/RueAdminService.java
package ma.zs.univ.service.facade.admin.commun;

import java.util.List;
import ma.zs.univ.bean.core.commun.Rue;
import ma.zs.univ.dao.criteria.core.commun.RueCriteria;
import ma.zs.univ.zynerator.service.IService;



public interface RueAdminService extends  IService<Rue,RueCriteria>  {

    List<Rue> findByQuartierId(Long id);
    int deleteByQuartierId(Long id);
    long countByQuartierCode(String code);




}

File: code/frontend/src/app/shared/service/admin/commun/RueAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {RueDto} from 'src/app/shared/model/commun/Rue.model';
import {RueCriteria} from 'src/app/shared/criteria/commun/RueCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class RueAdminService extends AbstractService<RueDto, RueCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): RueDto {
        return new RueDto();
    }

    public constrcutCriteria(): RueCriteria {
        return new RueCriteria();
    }

    get API() {
        return environment.apiUrlUnivservice + 'admin/rue/';
    }
}
